taca_ngi_pipeline/deliver/deliver_dds.py

```diff
@@ -0,0 +1,546 @@
+            self.pi_email = given_pi_email
+        if given_pi_name:
+            logger.warning("PI name for project {} specified by user: {}".format(self.projectid, given_pi_name))
+            self.pi_name = given_pi_name
+        if not self.pi_email and not self.pi_name:
+            try:
+                prj_order = self._get_order_detail()
+                self.pi_email = prj_order['fields']['project_pi_email']
+                self.pi_name = prj_order['fields']['project_pi_name']
+                logger.info("PI email for project {} found: {}".format(self.projectid, self.pi_email))
+                logger.info("PI name for project {} found: {}".format(self.projectid, self.pi_name))
+            except Exception as e:
+                logger.exception("Cannot fetch pi_email and/or name from StatusDB.")
+                raise e
+
+    def _set_other_member_details(self, other_member_emails=[], include_owner=False):
+        """Set other contact details if avilable, this is not mandatory so
+        the method will not raise error if it could not find any contact
+        """
+        self.other_member_details = []
+        # try getting appropriate contact emails
+        try:
+            prj_order = self._get_order_detail()
+            if include_owner:
+                owner_email = prj_order.get('owner', {}).get('email')
+                if owner_email and owner_email != self.pi_email and owner_email not in other_member_emails:
+                    other_member_emails.append(owner_email)
+            binfo_email = prj_order.get('fields', {}).get('project_bx_email')
+            if binfo_email and binfo_email != self.pi_email and binfo_email not in other_member_emails:
+                other_member_emails.append(binfo_email)
+        except (AssertionError, ValueError) as e:
+            pass # nothing to worry, just move on
+        if other_member_emails:
+            logger.info("Other appropriate contacts were found, they will be added to DDS delivery project: {}".format(", ".join(other_member_emails)))
+            self.other_member_details = other_member_emails
+
+    def _set_project_details(self, given_title=None, given_desc=None):
+        self.project_title, self.project_desc = (None, None)
+        if given_title:
+            logger.warning("Project title for project {} specified by user: {}".format(self.projectid, given_title))
+            self.project_title = given_title
+        if given_desc:
+            logger.warning("Project description for project {} specified by user: {}".format(self.projectid, given_desc))
+            self.project_desc = given_desc
+        if not self.project_desc or not self.project_title:
+            try:
+                prj_order = self._get_order_detail()
+                self.project_title = prj_order['title']
+                self.project_desc = prj_order['fields']['project_desc'].replace('\n', '\s')
```

Comment:
Not sure what `\s` does?

Reply:
Replace newline with space, but it's probably better to just use ' ' 😅 I'll change it 👍